internal/source/git.go

```diff
@@ -96,6 +113,18 @@ func (r *Git) Unpack(ctx context.Context, bundle *rukpakv1alpha1.Bundle) (*Resul
 	return &Result{Bundle: bundleFS, ResolvedSource: resolvedSource, State: StateUnpacked}, nil
 }
 
+func (r *Git) getCredential(ctx context.Context, bundle *rukpakv1alpha1.Bundle) (string, string, error) {
+	if bundle.Spec.Source.Git.SecretName == "" {
```

Comment:
Would it potentially make the logic more straightforward if this function were only called in the case where the `SecretName` is not blank? We could have something like the following:

```
if bundle.Spec.Source.Git.SecretName != "" {
    auth, err := r.getCredentials(ctx, bundle)
    if err != nil {
        return nil, err
    }
    cloneOpts.Auth = auth
}

Reply:
It's better.  I'll update it. Thanks!  I moved the `SecretName` check in the function for some reason in the PR for pre-golib refactor.  I don't remember the reason.  It may not be a valid reason any more after the refactor.